pkg/graveler/ref/manager.go

```diff
@@ -669,24 +691,51 @@ func (m *Manager) getPullWithPredicate(ctx context.Context, repository *graveler
 		return nil, nil, err
 	}
 	p := result.(*pullWithPred)
-	return p.PullRequest, p.Predicate, nil
+	return &p.PullRequest, p.Predicate, nil
 }
 
 func (m *Manager) GetPullRequest(ctx context.Context, repository *graveler.RepositoryRecord, pullID graveler.PullRequestID) (*graveler.PullRequest, error) {
 	pull, _, err := m.getPullWithPredicate(ctx, repository, pullID)
 	return pull, err
 }
 
+func (m *Manager) ListPullRequests(ctx context.Context, repository *graveler.RepositoryRecord) (graveler.PullsIterator, error) {
+	return NewPullsIterator(ctx, m.kvStore, repository)
+}
+
 func (m *Manager) CreatePullRequest(ctx context.Context, repository *graveler.RepositoryRecord, pullRequestID graveler.PullRequestID, pullRequest *graveler.PullRequest) error {
-	err := kv.SetMsgIf(ctx, m.kvStore, graveler.RepoPartition(repository), []byte(graveler.PullRequestPath(pullRequestID)), graveler.ProtoFromPullRequest(pullRequestID, pullRequest), nil)
+	// Save secondary index by source - dest. For now, we override the value. In the future we should allow only single src-dest to exist
```

Comment:
This means that the PR object is actually saved twice in the KV, one for the primary key and one for the secondary?

Reply:
No,
The secondary index contains the primary key.
Which means that reading the object via secondary key requires getting the primary key and then calling get using the primary key